client/layout/viewport-watcher/index.jsx

```diff
@@ -0,0 +1,40 @@
+/**
+ * External dependencies
+ */
+import { connect } from 'react-redux';
+import { debounce } from 'lodash';
+import React from 'react';
+import debugFactory from 'debug';
+
+/**
+ * Internal dependencies
+ */
+import { setViewportWidth } from 'state/ui/actions';
+import { getWindowInnerWidth } from 'lib/viewport';
+
+class ViewportWatcher extends React.Component {
+	componentWillMount() {
```

Comment:
Aside from the `global.window` check, I'm wondering if `componentDidMount` could be better, since it is never called in SSR scenarios.